homeassistant/helpers/device_registry.py

```diff
@@ -0,0 +1,124 @@
+"""Provide a way to connect entities belonging to one device."""
+import logging
+import uuid
+
+import attr
+
+from homeassistant.core import callback
+from homeassistant.loader import bind_hass
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_REGISTRY = 'device_registry'
+
+STORAGE_KEY = 'core.device_registry'
+STORAGE_VERSION = 1
+SAVE_DELAY = 10
+
+
+@attr.s(slots=True, frozen=True)
+class DeviceEntry:
+    """Device Registry Entry."""
+
+    identifiers = attr.ib(type=list)
+    manufacturer = attr.ib(type=str)
+    model = attr.ib(type=str)
+    connection = attr.ib(type=list)
+    name = attr.ib(type=str, default=None)
+    sw_version = attr.ib(type=str, default=None)
+    id = attr.ib(type=str, default=attr.Factory(lambda: uuid.uuid4().hex))
+
+
+class DeviceRegistry:
+    """Class to hold a registry of devices."""
+
+    def __init__(self, hass):
+        """Initialize the device registry."""
+        self.hass = hass
+        self.devices = None
+        self._sched_save = None
+        self._store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+
+    @callback
+    def async_get_device(self, identifiers: str, connections: tuple):
+        """Check if device is registered."""
+        for device in self.devices:
+            if any([identifier in device.identifiers
```

Comment:
No need for the `[` and `]`

Reply:
There where a few more places of self.devices that where needed to clean up :)